### ISSUE TITLE
API does not throw error while adding nodepool with AZ 

### ISSUE THREAD

**Issue by cpinjani** [rancher/aks-operator#751]
**Setup**
Rancher version: `v2.10-c4944b3ca0b39c474121df9e7ef31f7aff4e4f7a-head`
AKS Operator version: `v1.10.1-rc.2`

**Description**
Via UI, it is not possible to add nodepool with Availability Zone for a cluster created in region with no support for AZ. Updating this via API does not raise any error. It simply reverts the changes to AKSConfig after a few minutes.
The API should raise an error for the same.

**Steps to reproduce:**
- Create a cluster in region without AZ support Eg. WestUS
- Add nodepool via API and mention availability zones

**Expected results:**
The API must throw an error while adding nodepool with AZ in region where AZ are not supported.

**Logs**
```
time="2024-12-12T08:17:17Z" level=info msg="Checking configuration for cluster [cpinjani-akswest (id: c-4zdpb)]"
time="2024-12-12T08:17:18Z" level=info msg="Adding node pool [pool11] for cluster [cpinjani-akswest (id: c-4zdpb)]"
time="2024-12-12T08:17:24Z" level=info msg="Waiting for cluster [cpinjani-akswest (id: c-4zdpb)] to update node pool [pool11]"
time="2024-12-12T08:17:55Z" level=info msg="Checking configuration for cluster [cpinjani-akswest (id: c-4zdpb)]"
time="2024-12-12T08:17:56Z" level=info msg="Cluster [cpinjani-akswest (id: c-4zdpb)] finished updating"
time="2024-12-12T08:17:57Z" level=info msg="Checking configuration for cluster [cpinjani-akswest (id: c-4zdpb)]"
time="2024-12-12T08:17:59Z" level=info msg="Configuration for cluster [cpinjani-akswest (id: c-4zdpb)] was verified"
```

**PR's:**
- https://github.com/rancher/rancher/pull/48672
- https://github.com/rancher/rancher/pull/48716
- https://github.com/rancher/aks-operator/pull/836
- https://github.com/rancher/aks-operator/pull/830

**Comment by cpinjani** [comment]
**Test Result**
Rancher version: `v2.11-1d345932e5f78b3da07ebe13691c8224c62b6240-head`, AKS Operator version: `v1.11.0-rc.3`

Issue is reproducible on above build. Also the full validation is blocked by follow-up issue https://github.com/rancher/aks-operator/issues/850

cc: @mjura @mitulshah-suse

**Comment by rishabh625** [comment]
@cpinjani 
Even azure don't give any error , if you specify availability zones to nodepools , but cluster is created without mentioning az

**Comment by rishabhmsra** [comment]
Validated on rancher v2.11-head:

- Created AKS v1.32 cluster in WestUS region.
- Once cluster came into Active state, added a node pool mentioning the availability zones via API.
- Cluster goes into Updating state with following error on UI : `Failed to update cluster [rishabh-aks3 (id: c-zv2vj)]: availability zones are not supported in region westus`
- Since we getting error after sending request via API, closing this issue.
